File: Android/app/src/main/java/jp/co/riso/smartdeviceapp/model/printsettings/Group.java
/*
 * Copyright (c) 2014 RISO, Inc. All rights reserved.
 *
 * Group.java
 * SmartDeviceApp
 * Created by: a-LINK Group
 */

package jp.co.riso.smartdeviceapp.model.printsettings;

import java.util.ArrayList;
import java.util.List;

import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

/**
 * @class Group
 * 
 * @brief Group data class representing a group in PrintSettings. 
 * 
 * Data represented by \<group\> tag in XML.
 */
public class Group extends XmlNode {
    private List<Setting> mSettings;
    
    /**
     * @brief Creates a Group instance.
     * 
     * @param groupNode Node represented by \<group\> tag in XML.
     */
    public Group(Node groupNode) {
        super(groupNode);
        
        mSettings = new ArrayList<>();

        NodeList settingsList = groupNode.getChildNodes();
        for (int i = 1; i < settingsList.getLength(); i += 2) {
            mSettings.add(new Setting(settingsList.item(i)));    
        }
    }
    
    /**
     * @brief Gets settings of a Group.
     * 
     * @return List of Setting objects.
     */
    public List<Setting> getSettings() {
        return mSettings;
    }
}

File: Android/app/src/main/kotlin/jp/co/riso/smartdeviceapp/model/printsettings/Group.kt
/*
 * Copyright (c) 2022 RISO, Inc. All rights reserved.
 *
 * Group.kt
 * SmartDeviceApp
 * Created by: a-LINK Group
 */
package jp.co.riso.smartdeviceapp.model.printsettings

import org.w3c.dom.Node

/**
 * @class Group
 *
 * @brief Group data class representing a group in PrintSettings.
 *
 * Data represented by <group> tag in XML.
 */
class Group(groupNode: Node) : XmlNode(groupNode) {
    private val _settings: MutableList<Setting>

    /**
     * @brief Gets settings of a Group.
     *
     * @return List of Setting objects.
     */
    val settings: List<Setting>
        get() = _settings

    /**
     * @brief Creates a Group instance.
     *
     * @param groupNode Node represented by \<group></group>\> tag in XML.
     */
    init {
        _settings = ArrayList()
        val settingsList = groupNode.childNodes
        var i = 1
        while (i < settingsList.length) {
            _settings.add(Setting(settingsList.item(i)))
            i += 2
        }
    }
}